scripts/inference/convert_composer_to_hf.py

```diff
@@ -224,6 +381,83 @@ def main(args: Namespace) -> None:
         output_precision=args.output_precision,
         local_checkpoint_save_location=args.local_checkpoint_save_location)
 
+    dtype = {
```

Comment:
is there a reason you want to load the model back from disk to save it again? Or would it be possible to move this section into the `write_huggingface_pretrained_from_composer_checkpoint`?

Not blocking, doesn't really matter too much if this scripts takes 2min or 4min, but just curious if there's a technical reason for it.

Reply:
it was just to reduce moving parts while i tried to figure out how to edit the files, since i know `write_huggingface_pretrained_from_composer_checkpoint` works. In order for the files in the folder to be correct, you have to first register the classes, and then call `.save_pretrained` on the model, so it just cleaner conceptually to not add the mosaicgpt specific stuff to `write_huggingface_pretrained_from_composer_checkpoint`. that being said, i seem to have accidentally written the weights into the folder twice, so lemme figure that out.

Reply:
actually it would be a little bit tricky to combine, just because i want to make sure to go through the `from_pretrained` and `save_pretrained` interface for the model, and you can only call `from_pretrained` against a folder, but ill fix the double weight uploading.